source/ui/DisplayWindow.ooc

```diff
@@ -15,15 +15,60 @@
  * along with this software. If not, see <http://www.gnu.org/licenses/>.
  */
 
+use ooc-base
 use ooc-draw
 use ooc-geometry
 import x11/UnixWindow
 import win32/Win32DisplayWindow
+import GuiEvent
 
 DisplayWindow: abstract class {
+	_mousePressHandler: Event1<MouseEvent>
+	_mouseReleaseHandler: Event1<MouseEvent>
+	_keyPressHandler: Event1<KeyboardEvent>
+	_keyReleaseHandler: Event1<KeyboardEvent>
 	init: func (size: IntVector2D, title: String)
+	free: override func {
+		if (this _mousePressHandler)
+			this _mousePressHandler free()
+		if (this _mouseReleaseHandler)
+			this _mouseReleaseHandler free()
+		if (this _keyPressHandler)
+			this _keyPressHandler free()
+		if (this _keyReleaseHandler)
+			this _keyReleaseHandler free()
+		super()
+	}
 	draw: abstract func (image: Image)
 	refresh: virtual func
+	addMousePressHandler: func (handler: Event1<MouseEvent>) {
+		this _mousePressHandler = this _mousePressHandler ? this _mousePressHandler add(handler) : handler
+	}
+	addMouseReleaseHandler: func (handler: Event1<MouseEvent>) {
+		this _mouseReleaseHandler = this _mouseReleaseHandler ? this _mouseReleaseHandler add(handler) : handler
+	}
+	addKeyPressHandler: func (handler: Event1<KeyboardEvent>) {
+		this _keyPressHandler = this _keyPressHandler ? this _keyPressHandler add(handler) : handler
+	}
+	addKeyReleaseHandler: func (handler: Event1<KeyboardEvent>) {
+		this _keyReleaseHandler = this _keyReleaseHandler ? this _keyReleaseHandler add(handler) : handler
+	}
+	_onMousePress: func (event: MouseEvent) {
```

Comment:
These private methods will be used to pass the event to the window (for now from inside of event loop but it should be the work of yet non-existent `EventDispatcher`).